springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/components/DefaultComponentsScanner.java

```diff
@@ -17,28 +24,79 @@
 public class DefaultComponentsScanner implements ComponentsScanner {
 
     @Override
-    public Set<Class<?>> scanForComponents(Package basePackage) {
-        return scanForComponents(basePackage.getName());
+    public Set<Class<?>> scanForComponents(String basePackage) {
+        return this.scanForComponents(basePackage, null);
     }
 
     @Override
-    public Set<Class<?>> scanForComponents(String basePackage) {
-        log.debug("Scanning for component classes in {}", basePackage);
+    public Set<Class<?>> scanForComponents(String basePackage, String configurationBasePackage) {
+
+        Set<Class<?>> components;
+        if (StringUtils.isNotBlank(basePackage) && StringUtils.isNotBlank(configurationBasePackage)) {
+            log.debug("Scanning for component classes in configuration package {}, filtered by beans in {}", configurationBasePackage, basePackage);
```

Comment:
I think the filtering part is going to be too confusing - if the user want to scan based on beans in a configuration class, let it scan all the beans. Let's remove the filtering logic.